### PR TITLE
Create new game

### PR DESCRIPTION
**Pull request by radiofreemattd**
This commit does two things:

* It moves the controller and view logic that used to exist in `StaticPages#index` and moves that to `Games#index`. The thinking here is twofold - first, this seems like the proper place to list out the games and second, I imagine that we will want a static landing page that perhaps presents a prettifed version of an overview of the project.

* It allows the user to create a new game. Note that, for the sake of simplicity and shipping early, I made a decision that the player who creates the game will be the black player.

Let me know your thoughts and apologies for so many file changes...

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -3,12 +3,21 @@ class GamesController < ApplicationController
   before_action :authorize_player, only: :forfeit
```
Comment (jonmast) [inline app/controllers/games_controller.rb]: I believe the new and create actions should be authenticated as well.

**Comment by jonmast** [comment]
:+1: Looks good

File: spec/features/user_creates_new_game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    create(:game, black_player: user)
+
+    sign_in_as(user)
+    visit root_path
+    click_link 'Show me all games'
+    click_link 'Create a new game'
+    fill_in 'Name', with: 'TEST GAME - FEATURE TEST'
+    click_on 'Create a new game'
+
+    expect(page).to have_content('TEST GAME - FEATURE TEST')
```
Comment (mscoutermarsh) [inline spec/features/user_creates_new_game_spec.rb]: love it

File: spec/features/user_visits_games_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    expect(page).to have_css('.game')
+  end
+
+  scenario 'after clicking an individual game, user is on game page' do
+    game = create(:game, name: 'TEST GAME - FEATURE TEST')
+    visit root_path
+    click_link 'Show me all games'
+    click_link('TEST GAME - FEATURE TEST')
+    expect(current_path).to eq game_path(game)
```
Comment (mscoutermarsh) [inline spec/features/user_visits_games_index_spec.rb]: nice

**Pull request merged**